frontend-web/webclient/app/Applications/ApplicationsRow.tsx

```diff
@@ -1,34 +1,72 @@
 import {injectStyle} from "@/Unstyled";
 import {ApplicationGroup} from "./api";
-import {AppCard, ApplicationCardType} from "./Card";
+import {AppCard, AppCardType, AppCardStyle} from "./Card";
 import React, {useEffect, useState} from "react";
-import {Absolute, Flex, Grid, Icon, Link, Relative} from "@/ui-components";
+import {Absolute, Icon, Link, Relative} from "@/ui-components";
 import * as Pages from "./Pages";
+import {compute} from "@/UCloud";
+import ApplicationSummaryWithFavorite = compute.ApplicationSummaryWithFavorite;
+import {toggleAppFavorite} from "./Redux/Actions";
 
-const ApplicationRowContainerClass = injectStyle("tag-grid-bottom-box", k => `
+export const ApplicationRowContainerClass = injectStyle("application-row-container", k => `
     ${k} {
-        padding: 15px 10px 15px 10px;
+        display: flex;
+        justify-content: left;
+        gap: 20px;
+        padding: 12px 10px;
         margin: 0 -10px;
         overflow-x: auto;
     }
+
+    ${k}[data-space-between=true] {
+        justify-content: space-between;
+    }
 `);
 
+export interface ApplicationRowItem {
+    id: string,
+    type: AppCardType,
+    title: string,
+    description?: string,
+    defaultApplication?: string,
+    tags?: string[],
+    isFavorite?: boolean,
+    onFavorite?: (app: ApplicationSummaryWithFavorite) => void,
+    application?: ApplicationSummaryWithFavorite
+}
+
+export function ApplicationGroupToRowItem(group: ApplicationGroup): ApplicationRowItem {
+    return {
+        id: group.id.toString(),
+        type: AppCardType.GROUP,
+        title: group.title,
+        description: group.description,
+        defaultApplication: group.defaultApplication,
+        tags: group.tags
+    }
+}
+
+export function ApplicationSummaryToRowItem(app: ApplicationSummaryWithFavorite, onFavorite?: (app: ApplicationSummaryWithFavorite) => void): ApplicationRowItem {
+    return {
+        id: app.metadata.name,
+        type: AppCardType.APPLICATION,
+        title: app.metadata.title,
+        description: app.metadata.description,
+        isFavorite: app.favorite,
+        onFavorite: onFavorite,
```

Comment:
Why is the function saved in the object and not just passed to the same component?